sponsorship/src/test/java/com/kyriba/conference/sponsorship/api/PlanControllerTest.java

```diff
@@ -56,42 +60,56 @@ void registerPlan()
         .extract()
         .jsonPath()
         .get("id");
-    Assertions.assertNotNull(id);
+    assertNotNull(id);
   }
 
 
   @Test
   void cancelPlan()
   {
-    String id = given(specification)
+    given(specification)
         .contentType(MediaType.APPLICATION_JSON_UTF8_VALUE)
         .filter(document("/api/v1/sponsorship/plans/{id}/cancellation"))
         .when()
         .put("/api/v1/sponsorship/plans/123/cancellation")
         .then()
-        .statusCode(HttpStatus.SC_OK)
+        .statusCode(HttpStatus.SC_INTERNAL_SERVER_ERROR);
+  }
+
+
+  @Test
+  void getPlan404()
+  {
+    Object id = given(specification)
+        .contentType(MediaType.APPLICATION_JSON_UTF8_VALUE)
+        .filter(document("/api/v1/sponsorship/plans/{id}"))
+        .when()
+        .get("/api/v1/sponsorship/plans/404")
+        .then()
+        .statusCode(HttpStatus.SC_NOT_FOUND)
         .contentType(MediaType.APPLICATION_JSON_UTF8_VALUE)
         .extract()
         .jsonPath()
         .get("id");
-    Assertions.assertNotNull(id);
+    assertNull(id);
   }
 
 
   @Test
+  @Sql("/test_data_plan.sql")

```

Comment:
Probably you could easily create plan via API and not break "black box".

Reply:
Will be updated in the next pull request with unit tests